test/language/statements/class/subclass/private-class-field-on-nonextensible-return-override.js

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2019 Caio Lima. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: It is not possible to add private fields on non-extensible objects via return override
+esid: sec-define-field
+info: |
+  1.1 PrivateFieldAdd ( O, P, value )
+    1. If O.[[Extensible]] is false, throw a TypeError exception.
+    ...
+
+features:
+  - class
+  - class-fields-private
+  - class-fields-public
+  - nonextensible-applies-to-private
+flags: [onlyStrict]
+---*/
+
+// Analogous to
+// test/language/statements/class/elements/private-class-field-on-nonextensible-objects.js
+
+class TrojanBase {
+  constructor(obj) {
+    return obj;
+  }
+}
+
+
+// extend superclass with private instance data field
+class ClassWithPrivateField extends TrojanBase {
+  #val;
+
+  constructor(obj) {
+    super(obj);
+    this.#val = 42;
+  }
+  val() {
+    return this.#val;
+  }
+}
+
+const t = new ClassWithPrivateField({});
+// extensible objects can be extended
+assert.sameValue(t.val(), 42);
```

Comment:
This looks incorrect. You expect `t` to have `val` method, but `t` is `{}` i.e. does not have methods from `ClassWithPrivateField.prototype`.

Reply:
In progress at https://github.com/tc39/test262/pull/4590